### ISSUE TITLE
Markdown Only PRs blocked by required builds

### ISSUE THREAD

**Issue by TimothyMothra** [open-telemetry/opentelemetry-dotnet#3106]
I've got some PRs updating docs and I've noticed that not all the builds run.

<img width="676" alt="image" src="https://user-images.githubusercontent.com/28785781/160723433-98649bdd-2523-4e82-a408-2795132c34a6.png">

I think this is due to this line:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/add7eee59cb86fafec1896dd5ce54714348b5ae4/.github/workflows/windows-ci.yml#L10-L11

GitHub suggests having alternate workflows with the same name to work around this.
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks

> If someone submits a pull request that changes a markdown file in the root of the repository, then the workflow above won't run at all because of the path filtering. As a result you won't be able to merge the pull request. You would see the following status on the pull request:
You can fix this by creating a generic workflow, with the same name, that will return true in any case similar to the workflow below :

--

## Scope
- [ ] apicompatibility (not required)
- [ ] code-coverage (not required)
- [x] integration #3132
- [x] linux-ci #3128
- [x] windows-ci #3116 
- [x] dotnet-format #3133

**Comment by TimothyMothra** [comment]
FYI @cijothomas 

I'll start with the windows-ci and if it works as expected, will expand to the others.

**Comment by TimothyMothra** [comment]
I tested a few iterations of this and my first PR is what I think is the best version.

For normal PRs (not documentation), I tried excluding this workflow but wasn't successful.
I verified it's not detrimental to have this run in addition to required workflows.
However it does make the list of Checks a little bit confusing with duplicate names.

After this first PR merges, I proposing that we wait a few days before adding the other workflows.
I'll check new PRs and verify that nothing is blocked/broken by adding this workflow.
I also want to confirm that having the duplicate checks doesn't irritate our maintainers before adding additional checks.
